adapters/IronSource/IronSourceAdapter/GADMAdapterIronSourceRtbInterstitialAd.m

```diff
@@ -0,0 +1,193 @@
+                         completionHandler:
+(GADMediationInterstitialLoadCompletionHandler)completionHandler {
+    _interstitalAdLoadCompletionHandler = completionHandler;
+    
+    NSDictionary *credentials = [adConfiguration.credentials settings];
+    NSString *applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    
+    if (applicationKey != nil && ![GADMAdapterIronSourceUtils isEmpty:applicationKey]) {
+        applicationKey = credentials[GADMAdapterIronSourceAppKey];
+    } else {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorInvalidServerParameters,
+                                                                          @"Missing or invalid IronSource application key.");
+        
+        _interstitalAdLoadCompletionHandler(nil, error);
+        return;
+    }
+    
+    if (credentials[GADMAdapterIronSourceInstanceId]) {
+        self.instanceID = credentials[GADMAdapterIronSourceInstanceId];
+    } else {
+        [GADMAdapterIronSourceUtils onLog:@"Missing or invalid IronSource interstitial ad Instance ID. "
+         @"Using the default instance ID."];
+        self.instanceID = GADMIronSourceDefaultRtbInstanceId;
+    }
+    
+    NSString *bidResponse = adConfiguration.bidResponse;
+    NSMutableDictionary<NSString *, NSString *> *extraParams= [GADMAdapterIronSourceUtils getExtraParamsWithWatermark:adConfiguration.watermark];
+    
+    ISAInterstitialAdRequest *adRequest = [[[[ISAInterstitialAdRequestBuilder alloc] initWithInstanceId: self.instanceID adm: bidResponse]withExtraParams:extraParams] build];
+    
+    [ISAInterstitialAdLoader loadAdWithAdRequest: adRequest delegate: self];
+}
+
+#pragma mark - GADMediationInterstitialAd
+
+- (void)presentFromViewController:(nonnull UIViewController *)viewController {
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"Showing IronSource interstitial ad for Instance ID: %@",
+            self.instanceID]];
+    id<GADMediationInterstitialAdEventDelegate> interstitialDelegate = self.interstitialAdEventDelegate;
+
+    if (!self.biddingISAInterstitialAd){
+        if (interstitialDelegate){
+            NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                              GADMAdapterIronSourceErrorFailedToShow,
+                                                                              @"the ad is nil");
+            [interstitialDelegate didFailToPresentWithError:error];
+        }
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString stringWithFormat:@"Failed to show due to ad not loaded, for Instance ID: %@",
+                self.instanceID]];
+        return;
+    }
+    
+    [self.biddingISAInterstitialAd setDelegate:self];
+    [self.biddingISAInterstitialAd showFromViewController:viewController ];
+}
+
+
```

Comment:
Should add pragma mark ISAInterstitialAdDelegate and ISAInterstitialAdLoaderDelegate for the methods below this line.

Reply:
Done